stripe-ui-core/src/main/java/com/stripe/android/uicore/navigation/NavigationEffects.kt

```diff
@@ -0,0 +1,78 @@
+package com.stripe.android.uicore.navigation
+
+import android.app.Activity
+import androidx.annotation.RestrictTo
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.navigation.NavBackStackEntry
+import androidx.navigation.NavHostController
+import androidx.navigation.NavOptionsBuilder
+import androidx.navigation.compose.currentBackStackEntryAsState
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
+
+@Composable
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+fun NavigationEffects(
+    navigationChannel: SharedFlow<NavigationIntent>,
```

Comment:
Should we use this to convert this from a `SharedFlow` to a `StateFlow`, so we avoid losing events?

Reply:
I like the idea! let's do it on a separate PR so we can independently test it from the migration?

Reply:
Yep, just wanted it to be on our radar 👍